src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -147,6 +147,10 @@ class CoreWorkerDirectTaskSubmitter {
   /// if a remote raylet tells us to spill the task back to the local raylet.
   const ClientID local_raylet_id_;
 
+  /// The gcs client to communicate with gcs server. When GCS-Based actor management is
+  /// ready, the actor creation needs go through gcs server.
+  std::shared_ptr<gcs::GcsClient> gcs_client_;
```

Comment:
> It would be good to only give this class a reference to the Actors() interface instead of the entire GCS client, to make unit testing easier.

Good suggestion, maybe better to give an  anonymous function named `create_actor_helper` which invoke `gcs_client_->Actors().AsyncCreateActor(task_spec, callback)` outside the `CoreWorkerDirectTaskSubmitter`.

Reply:
I'm not sure if I understand your suggestion.

I'm suggesting that we do
`std::shared_ptr<ActorInfoAccessor> actor_table_;`
instead of
`std::shared_ptr<gcs::GcsClient> gcs_client_;`

Reply:
> I'm not sure if I understand your suggestion.
> 
> I'm suggesting that we do
> `std::shared_ptr<ActorInfoAccessor> actor_table_;`
> instead of
> `std::shared_ptr<gcs::GcsClient> gcs_client_;`

This would also make unit testing much easier. Right now, the code brings in a huge dependency since it requires the entire GcsClient. But we only need the actors interface, right?